[ENH] add preregistration URL to dataset description (#572)
bids-standard/bids-specification: src/03-modality-agnostic-files.md

| EthicsApprovals    | OPTIONAL. List of ethics committee approvals of the research protocols and/or protocol identifiers.                                                                                                                                                                |
| ReferencesAndLinks | OPTIONAL. List of references to publication that contain information on the dataset, or links.                                                                                                                                                                     |
| DatasetDOI         | OPTIONAL. The Document Object Identifier of the dataset (not the corresponding paper).                                                                                                                                                                             |

> | Preregistration    | OPTIONAL.  URL or Document Object Identifier of the time-stamped pre-registration of the study that produced the dataset.                                                                                                                                          |

Reviewer: DOIs can be made URIs by prepending `doi:` (or URLs by prepending `https://doi.org/`), so this puts the address of one URI type on par with a full URI. An implementing tool would basically have to say:

```
try:
    val = parseurl(preregistration)
except:
    val = parseuri("doi:" + preregistration)
```

It seems conceptually cleaner to permit any URI:

```suggestion
| Preregistration    | OPTIONAL. URI of a time-stamped preregistration of the study that produced the dataset.                                                                                                                                                                            |
```

I'm not sure if there's an argument for narrowing to URL.

I am also :+1: for adding an example URI to add to the `dataset_description.json`.

Reviewer 2: a `study that produced the dataset` sounds weird, people produce datasets
suggest
`URL or DOI of a time-stamped preregistration of the study from which the dataset originates.`

    "Alzheimer A., & Kraepelin, E. (2015). Neural correlates of presenile dementia in humans. Journal of Neuroscientific Data, 2, 234001. doi:1920.8/jndata.2015.7"
  ],
  "DatasetDOI": "doi:10.0.2.3/dfjj.10",
  "HEDVersion": "7.1.1",

>   "Preregistration": "doi:10.1038/s41598-018-30701-5"

Reviewer 3: This is the eventual article, it seems ? I'd expect that under "ReferencesAndLinks". I'd think to use this preregistration: https://osf.io/uf9gh

Reviewer 3: I notice that the prereg itself doesn't have a DOI, which is maybe what influenced the decision ?

Still, I'd think we'd want to point to the original prereg in this field, if at all possible...

Author: Oh dear. :facepalm: 
I did rush too much on this one. 

Thanks @emdupre for catching it. And yeah you are right this is exactly how I mixed things up: I assumed the DOI on the pre-reg page was that of the pre-reg and not the article.
